src/instructionFiles/InstructionFileManager.ts

```diff
@@ -0,0 +1,311 @@
+        let content = fs.readFileSync(extensionFilePath, 'utf8');
+        content = content.replace('{{VERSION}}', this.CURRENT_VERSION.toString());
+        return content;
+      }
+
+      // Fallback content
+      return `# NPL Development v${this.CURRENT_VERSION}\n\nWhen working with NPL files:\n\n1. NPL is a domain-specific language\n2. Follow existing code style\n\n${this.NPL_SECTION_END}`;
+    } catch (error) {
+      console.error(`Error loading template: ${error}`);
+      return `# NPL Development v${this.CURRENT_VERSION}\n\nWhen working with NPL files:\n\n1. NPL is a domain-specific language\n2. Follow existing code style\n\n${this.NPL_SECTION_END}`;
+    }
+  }
+
+  private async createInstructionFile(filePath: string, templatePath: string): Promise<void> {
+    const dirPath = path.dirname(filePath);
+
+    if (!fs.existsSync(dirPath)) {
+      fs.mkdirSync(dirPath, { recursive: true });
+    }
+
+    fs.writeFileSync(filePath, this.getTemplateContent(templatePath), 'utf8');
+  }
+
+  private async appendNplSection(filePath: string, existingContent: string, templatePath: string): Promise<void> {
+    const newContent = `${existingContent.trim()}\n\n${this.getTemplateContent(templatePath)}`;
+
+    fs.writeFileSync(filePath, newContent, 'utf8');
+  }
+
+  private async updateNplSection(filePath: string, existingContent: string, templatePath: string): Promise<void> {
+    const newContent = this.replaceNplSection(existingContent, templatePath);
+
+    fs.writeFileSync(filePath, newContent, 'utf8');
+  }
+
+  private replaceNplSection(content: string, templatePath: string): string {
+    const startRegex = new RegExp(`${this.NPL_SECTION_START}\\d+`);
+    const startMatch = content.match(startRegex);
+    if (!startMatch) {
+      return content;
+    }
+
+    const startIndex = startMatch.index;
+    if (startIndex === undefined) {
+      return content;
+    }
+
+    const endIndex = content.indexOf(this.NPL_SECTION_END, startIndex);
+    if (endIndex === -1) {
+      return `${content}\n\n${this.getTemplateContent(templatePath)}`;
+    }
+
+    const endOfSection = endIndex + this.NPL_SECTION_END.length;
+    return content.slice(0, startIndex) + this.getTemplateContent(templatePath) + content.slice(endOfSection);
+  }
+
+  private getTemplatePath(): string {
+    if (extensionContext) {
+      // Production path
+      const prodPath = path.join(extensionContext.extensionPath, 'out', RESOURCES_DIR, TEMPLATES_DIR, NPL_INSTRUCTIONS_TEMPLATE_FILENAME);
+      if (fs.existsSync(prodPath)) {
+        return prodPath;
+      }
+
+      // Root resources path
+      const altPath = path.join(extensionContext.extensionPath, RESOURCES_DIR, TEMPLATES_DIR, NPL_INSTRUCTIONS_TEMPLATE_FILENAME);
+      if (fs.existsSync(altPath)) {
+        return altPath;
+      }
+    }
+
+    // Fallback relative path
+    return path.join(__dirname, '..', RESOURCES_DIR, TEMPLATES_DIR, NPL_INSTRUCTIONS_TEMPLATE_FILENAME);
```

Comment:
This stuff is horrible, but I'm not sure how to do this cleanly 😢 – js test/prod paths are black magic to me right now